assets/js/blocks/product-query/inspector-controls/product-selector.tsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import { getProducts } from '@woocommerce/editor-components/utils';
+import { ProductResponseItem } from '@woocommerce/types';
+import { objectOmit } from '@woocommerce/utils';
+import { useState, useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import {
+	FormTokenField,
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalToolsPanelItem as ToolsPanelItem,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { ProductQueryBlock } from '../types';
+import { setQueryAttribute } from '../utils';
+
+export const ProductSelector = ( props: ProductQueryBlock ) => {
```

Comment:
I am nitpicking here, but how do you feel about restructuring the attributes property from the props object to simplify property access:
```suggestion
export const ProductSelector = ({ attributes }: ProductQueryBlock) => {
```

Reply:
Thanks for the suggestion! Normally, I'd be up for it, but in this case you can see that we use `props` as a whole twice on L76 and L85.